src/index.ts

```diff
@@ -229,21 +270,23 @@ class FSMerger {
       }
     }
     if (!existingPath.length) {
-      fs.readdir(fullDirPath, options, callback);
+      nodefs.readdir(fullDirPath, options, callback);
     }
     let readComplete = 0;
     for (let i = 0; i < existingPath.length; i++) {
-      fs.readdir(existingPath[i], options, (err: NodeJS.ErrnoException | null , list: string[]) => {
+      nodefs.readdir(existingPath[i], <any>options, (err: NodeJS.ErrnoException | null , list: string[]) => {
         readComplete += 1;
-        result && result.push.apply(result, list);
+        if (list) {
+          for (let r of list) {
+            result.add(r);
+          }
+        }
```

Comment:
this used to guard on `result`, which is always true. I think you meant to guard on `list` (which would be undefined in the case of an error).

FYI: I'm using a single set in this code here because making a new set object with each iteration seems like a lot of extra work for the computer.